sound/soc/sof/sof-pci-dev.c

```diff
@@ -21,6 +21,21 @@
 #include "intel/shim.h"
 #include "intel/hda.h"
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_APOLLOLAKE) ||\
+	IS_ENABLED(CONFIG_SND_SOC_SOF_GEMINILAKE)
+static const struct sof_intel_dsp_desc apollolake_chip_info = {
+	/* Apollolake */
+	.cores_num = 2,
+	.cores_mask = HDA_DSP_CORE_MASK(0) | HDA_DSP_CORE_MASK(1),
+	.ipc_req = HDA_DSP_REG_HIPCI,
+	.ipc_req_mask = HDA_DSP_REG_HIPCI_BUSY,
+	.ipc_ack = HDA_DSP_REG_HIPCIE,
+	.ipc_ack_mask = HDA_DSP_REG_HIPCIE_DONE,
+	.ipc_ctl = HDA_DSP_REG_HIPCCTL,
+	.ops = &sof_apl_ops,
+};
+#endif
```

Comment:
Ah sorry, there was a misunderstanding here. Can we move all these chip_info tables to the hda.c file (same as before)? It's not really cool to pollute sof-pci-dev.c will all these intel-specific definitions. The _desc tables can stay here.

Reply:
yes, that why I ask you whether sof-pci-dev.c is a generic one or not. I will refine it.